Sources/StreamChat/Database/DTOs/MessageDTO.swift

```diff
@@ -348,13 +346,22 @@ extension NSManagedObjectContext: MessageDatabaseSession {
         } else {
             log.assertionFailure("Should never happen because either `cid` or `payload.channel` should be present.")
         }
+        
+        if let lastMessageAt = channelDTO?.lastMessageAt {
+            channelDTO?.lastMessageAt = max(lastMessageAt, payload.createdAt)
+        }
 
         if let channelDTO = channelDTO {
+            if let lastMessageAt = channelDTO.lastMessageAt {
+                channelDTO.lastMessageAt = max(lastMessageAt, payload.createdAt)
+            }
```

Comment:
we need to update the `channelDTO.lastMessageAt` also when the current `lastMessageAt` is `nil`